lib/bosh/BoshDirectorClient.js

```diff
@@ -405,9 +406,12 @@ class BoshDirectorClient extends HttpClient {
             }
           })
           .catch(err => {
+            errorRetries++;
             logger.error(`error occurred while fetching state of task id: ${taskId}`, err);
```

Comment:
May be we can include errorRetries (the count in present loop) in log..... so in a huge log file it will be easy to debug.